octobot_trading/data/order.py

```diff
@@ -254,38 +264,67 @@ def is_self_managed(self):
 
     def update_from_raw(self, raw_order):
         if self.side is None or self.order_type is None:
-            self.__update_type_from_raw(raw_order)
-            if self.taker_or_maker is None:
-                self.__update_taker_maker_from_raw()
+            try:
+                self.__update_type_from_raw(raw_order)
+                if self.taker_or_maker is None:
+                    self.__update_taker_maker_from_raw()
+            except KeyError:
+                pass
 
         return self.update(**{
-            "symbol": raw_order[ExchangeConstantsOrderColumns.SYMBOL.value],
-            "current_price": raw_order[ExchangeConstantsOrderColumns.PRICE.value],
-            "quantity": raw_order[ExchangeConstantsOrderColumns.AMOUNT.value],
-            "price": raw_order[ExchangeConstantsOrderColumns.PRICE.value],
+            "symbol": get_value_or_default(raw_order, ExchangeConstantsOrderColumns.SYMBOL.value, None),
+            "current_price": get_value_or_default(raw_order, ExchangeConstantsOrderColumns.PRICE.value, None),
+            "quantity": get_value_or_default(raw_order, ExchangeConstantsOrderColumns.AMOUNT.value, None),
+            "price": get_value_or_default(raw_order, ExchangeConstantsOrderColumns.PRICE.value, None),
             "stop_price": None,
-            "status": OrderStatus(raw_order[ExchangeConstantsOrderColumns.STATUS.value]),
-            "order_id": raw_order[ExchangeConstantsOrderColumns.ID.value],
-            "quantity_filled": raw_order[ExchangeConstantsOrderColumns.FILLED.value],
-            "filled_price": raw_order[ExchangeConstantsOrderColumns.PRICE.value],
-            "total_cost": raw_order[ExchangeConstantsOrderColumns.COST.value],
-            "fee": raw_order[ExchangeConstantsOrderColumns.FEE.value],
-            "timestamp": raw_order[ExchangeConstantsOrderColumns.TIMESTAMP.value]
+            "status": Order.parse_order_status(raw_order),
+            "order_id": get_value_or_default(raw_order, ExchangeConstantsOrderColumns.ID.value, None),
+            "quantity_filled": get_value_or_default(raw_order, ExchangeConstantsOrderColumns.FILLED.value, None),
+            "filled_price": get_value_or_default(raw_order, ExchangeConstantsOrderColumns.PRICE.value, None),
+            "total_cost": get_value_or_default(raw_order, ExchangeConstantsOrderColumns.COST.value, None),
+            "fee": get_value_or_default(raw_order, ExchangeConstantsOrderColumns.FEE.value, None),
+            "timestamp": get_value_or_default(raw_order, ExchangeConstantsOrderColumns.TIMESTAMP.value, None)
         })
 
-    def __update_type_from_raw(self, raw_order):
-        self.side: TradeOrderSide = TradeOrderSide(raw_order[ExchangeConstantsOrderColumns.SIDE.value])
+    @staticmethod
+    def parse_order_type(raw_order):
```

Comment:
I guess it's a good time to integrate new order types from https://github.com/Drakkar-Software/OctoBot/commit/3160e197bd841a49e2e9b54a43bf4fa13f03bb02

Reply:
The new order type have already been integrated in this folder : https://github.com/Drakkar-Software/OctoBot-Trading/tree/master/octobot_trading/orders (trailing stop etc...). Are you talking about that ?

Reply:
Right, but to my understanding, if we have a stop profit type order from exchange for example , it won't be recognized in this method (ex if order_type == TradeOrderType.TAKE_PROFIT). Or maybe you mean it's not a problem ?

Reply:
Oh yeah you are right :+1:

Reply:
Actually the problem come from CCXT. CCXT is giving us the side of the order (BUY or SELL) and a MARKET or a LIMIT. There is currently no possibility to have something else. Except for private methods implemented in specific exchanges that shouldn't be supported here I think.

Reply:
The code in https://github.com/Drakkar-Software/OctoBot/commit/3160e197bd841a49e2e9b54a43bf4fa13f03bb02 ex: ```order_type == TradeOrderType.TAKE_PROFIT_LIMIT``` was working for me with binance new order types with real trader, without any specific code, maybe it's not broken ?